modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/QueryCancelledException.java

```diff
@@ -33,4 +34,16 @@ public class QueryCancelledException extends IgniteException {
     public QueryCancelledException() {
         super(ERR_MSG);
     }
+
+    /**
+     * Creates a new exception with the given trace id, error code, detail message and cause.
+     *
+     * @param traceId Unique identifier of this exception.
+     * @param code Full error code.
+     * @param message Detail message.
+     * @param cause Optional nested exception (can be {@code null}).
+     */
+    public QueryCancelledException(UUID traceId, int code, String message, Throwable cause) {
```

Comment:
I suppose we shouldn't have a rule to have specific constructors for internal exceptions, just for public one.
In case we throws internal exception throught public API, we should fix it instead of cover up the problem

Reply:
Good point, fixed: 
* Reverted internal class changes.
* Updated the test to skip classes in internal packages.